pkg/skaffold/initializer/init.go

```diff
@@ -150,94 +172,136 @@ func DoInit(out io.Writer, c Config) error {
 	return nil
 }
 
-func processCliArtifacts(artifacts []string) ([]dockerfilePair, error) {
-	var pairs []dockerfilePair
+// autoSelectBuilders takes a list of builders and images, checks if any of the builders' configured target
+// images match an image in the image list, and returns a list of the matching builder/image pairs. Also
+// returns the images from the original image list that didn't match any build configurations.
```

Comment:
I think at some point I was passing a pointer and modifying the original list, then forgot to return the filtered list when I changed it to pass a copy instead of a pointer, whoops.